### DISCUSSION TITLE
How far is version 1.0 (due to breaking changes in 0.12)?

### DISCUSSION THREAD

**Discussion by smainz** [jwtk/jjwt#900] [Q&A] [answered]
Hello!

Thank you very much for this great library. We have been using it for some time now. 

Since the release of version 0.12, our Renovate bot has been bugging us to update the dependency. Unfortunately, the release notes for 0.12 say: 

> This is our first-ever breaking change release. While we tried hard to minimize the breakages, some were just necessary in preparation for 1.0 and to finalize all JWT RFC features. If you are not partial to fixing changes when upgrading a library, we strongly encourage you to wait until the 1.0 release.

If we are close to version 1.0, I would wait, but if it is still a long way off, we need to take care of the changes now.

Do you have a rough estimate of when version 1.0 is due?

Many thanks in advance

--
I think I need to ask this question here, not on Stack Overflow as it is stated in your issue template. If you think the question does not belong here, please close the issue.

**Comment by lhazlewood** [comment, marked as answer]
Hi @smainz !  I migrated the issue to the GitHub Discussions feature, as that's what we use for general questions and/or conversations that aren't necessarily directly attributed to changing code.

As for a 1.0 timeline, it's probably a decent ways away (at least 6 months or more) because we want to also fully support non-compact (standard JSON, non-compact JWT) parsing as well.

Most people don't really utilize this feature of the various JOSE RFC specifications (where you can sign or encrypt full JSON documents, not just compact 'tokens'), but that is the next/final last 'big thing' for JJWT.   Once that's done, we'll be fully RFC compliant, and we wanted that to coincide with a 1.0 release.

I hope that helps!

**Comment by smainz** [comment]
Thank you, that helps.

We will start updating our code now for 0.12.
